core/src/main/java/org/commonjava/indy/core/ctl/NfcController.java

```diff
@@ -80,7 +98,19 @@ public NotFoundCacheDTO getAllMissing()
     }
 
     public NotFoundCacheDTO getMissing( final StoreKey key )
-        throws IndyWorkflowException
+                    throws IndyWorkflowException
+    {
+        return doGetMissing( key );
+    }
+
+    public NotFoundCacheDTO getMissing( final StoreKey key, int pageIndex, int pageSize )
+                    throws IndyWorkflowException
+    {
+        return doGetMissing( key, pageIndex, pageSize );
+    }
+
+    private NotFoundCacheDTO doGetMissing( final StoreKey key, int... pagingParams )
```

Comment:
IMO we should establish a pattern to use for paging here, then we can reuse it for things like retrieving the list of RemoteRepositories or retrieving the list of Groups, etc. Ideally, we should have a Paging class that contains the fields so the code is a little more self-documenting and less likely to be subtly wrong. It would also allow us to provide defaults or null values if necessary.

Reply:
It sounds a good idea. How about open another issue to track? This may need some investigation ( borrow some idea from somewhere else ).

Reply:
Done.